python/lsst/pipe/supertask/super_task.py

```diff
@@ -101,119 +44,72 @@ class SuperTask(Task):
+                            self._DefaultName
         :param parent_task: the parent task of this subtask, if any.
                              - If None (a top-level task) then you must specify config and name is
                                ignored.
                              - If not None (a subtask) then you must specify name
-        :param log:         pexLog log; if None then the default is used;
-                            in either case a copy is made using the full task name.
-        :param activator:   name of activator calling this task, default is None,
-                            there is only one kind of activator (cmdLineActivator)
+        :param log:         log (an lsst.log.Log) whose name is used as a log name prefix,
+                            or None for no prefix. Ignored if parentTask specified, in which case
+                            parentTask.log's name is used as a prefix.
+        :param butler:      data butler instance (this is not used by this class, it should
+                            probably be removed)
         :return: The SuperTask Class
         """
 
+        # No spaces allowed in names for SuperTasks
+        # TO DO: need better validation here, some other characters may be disallowed
         if name is None:
-            name = getattr(self, "_default_name", None)
+            name = getattr(self, "_DefaultName", None)
+        if name is not None:
+            name = name.replace(" ", "_")
         super(SuperTask, self).__init__(config, name, parent_task, log)  # Initiate Task
 
-        self._parser = None
-        self.input = Struct()  # input and output will be deprecated
-        self.output = None
-        self._activator = activator
         self._completed = False  # Hook to indicate whether a SuperTask was completed
-        self.list_config = None  # List of configuration items
-        self.name = self.name.replace(" ", "_")  # No spaces allowed in names for SuperTasks
+        self._list_config = None  # List of configuration items
         self._task_kind = 'SuperTask'  # To differentiate between this and Task
 
         self.log.info('%s was initiated' % self.name)  # For debugging only,  to be removed
 
     @property
     def name(self):
         """
-        name of task as property
-        :return: Name of (Super)Task
-        """
-        return self._name
-
-    @name.setter
-    def name(self, name):
-        """
-        Sets the name of task
-        :param name: new name for (Super)Task
-        """
-        self._name = name
-
-    @property
-    def activator(self):
-        """
-        Hook for the name of activator as a property
-        :return: Name of activator invoking this (Super)Task
-        """
-        return self._activator
+        Name of this super-task.
 
-    @activator.setter
-    def activator(self, activator):
+        Note that base class defines `getName()` method which returns the same information,
```

Comment:
I'll likely remove `name` property and replace its use with `getName()` everywhere, again for consistency with existing base class interface.